### PR TITLE
Tariff reallocator ut

### PR DESCRIPTION
**Pull request by iaincrawf**
Ripped the products solution stuff out, so now is front end and then makes a post request to products solution. Also change of terminology from sales articles reallocate to tariff reallocate.

- Can't get errors to appear in the browser properly anymore eg when I return a bad request or not authorised - in redux it seems to be putting the error into errors -> request errors instead of errors -> item errors . I thought that'd be something reducer-y or action-y but I can't see anything different from anywhere else, or anything in the back end that's wrong

- Haven't got post request to products right yet, haven't spent a lot of time on that yet though so still got a few slightly different calls I think I can try.

If anyone has an idea on fixing either of the above pls let me know!

### CODE REVIEW

File: src/Facade/StyleCop.Cache [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        <warning>False</warning>
+      </violation>
+      <violation namespace="StyleCop.CSharp.DocumentationRules" rule="ElementsMustBeDocumented" ruleCheckId="SA1600">
+        <context>The method must have a documentation header.</context>
+        <line>11</line>
+        <warning>False</warning>
+      </violation>
+    </violations>
+  </sourcecode>
+</stylecopresultscache>
```
Comment (lewisrenfrew) [inline src/Facade/StyleCop.Cache:46]: Is this file supposed to be commited?

File: src/Service.Host/client/src/containers/tariffs/Reallocator.js [hunk 2]
```diff
@@ -3,42 +3,40 @@ import {
...
         .map(s => ({ ...s, id: s.id, name: s.tariffCode, description: s.description })),
     oldTariffSearchErrorMessage: getItemErrorDetailMessage(state, itemTypes.oldTariffs.item),
-    item: salesArticlesReallocateSelectors.getItem(state)
+    item: tariffsReallocateSelectors.getItem(state)
 });
 
-const initialise = () => dispatch => {
-    dispatch(salesArticleReallocateActions.fetch());
-};
+const initialise = () => () => {};
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/containers/tariffs/Reallocator.js:32]: would just ditch this if it's not doing anything and remove the initialiseOnMountCall too

File: src/Service.Host/client/src/components/Root.js [hunk 3]
```diff
@@ -71,7 +71,7 @@ import SalesArticlesByTariff from '../containers/reports/SalesArticlesByTariff';
 import SalesArticleTriggerLevels from '../containers/reports/SalesArticleTriggerLevels';
 import WeeeReportOptions from '../containers/reportOptions/WeeeReportOptions';
 import WeeeReport from '../containers/reports/WeeeReport';
-import SalesArticleReallocator from '../containers/salesArticles/Reallocator';
+import tariffsReallocator from '../containers/tariffs/Reallocator';
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/components/Root.js:75]: Components should be capitalized (TariffsReallocator)

File: src/Service.Host/client/src/itemTypes.js [hunk 4]
```diff
@@ -46,10 +46,10 @@ export const salesArticles = new ItemType(
     'SALES_ARTICLES',
     '/products/maint/sales-articles'
 );
-export const salesArticlesReallocate = new ItemType(
-    'salesArticlesReallocate',
-    'SALES_ARTICLES_REALLOCATE',
-    '/products/maint/sales-articles-reallocate'
+export const TariffsReallocate = new ItemType(
```
Comment (lewisrenfrew) [inline src/Service.Host/client/src/itemTypes.js:49]: this should have a lowercase t as in tariffsReallocate. 

itemType names need to perfectly match the reducer names for a few things to work correctly.

If you fix that (and update any references of course) then that should solve your error messages problem.

**Review by lewisrenfrew — APPROVED** [review]

File: src/Service.Host/client/src/itemTypes.js (hunk 4 above)
Comment (iaincrawf) [reply src/Service.Host/client/src/itemTypes.js]: Thank you!!!! Think I was just too late in the day to spot the capital there, that of course did allow me to get the errors displaying properly :)

File: src/Facade/StyleCop.Cache (hunk 1 above)
Comment (iaincrawf) [reply src/Facade/StyleCop.Cache]: No idea must've missed it slipping in to a commit, have deleted it

**Pull request merged**